web/user_profile/views.py

```diff
@@ -145,3 +145,19 @@ def put(self, request, *args, **kwargs):
 
     def perform_create(self, serializer):
         serializer.save(id=self.kwargs['user_id'])
+
+
+class ProfileSettingsRetrieveViewSet(viewsets.GenericViewSet):
+    template_name = "profile-settings.html"
+    serializer_class = serializers.ProfileUpdateSerializer
+    queryset = User.objects.all()
+
+    def get_object(self):
+        queryset = self.filter_queryset(self.get_queryset())
+        obj = queryset.get(pk=self.kwargs['user_id'])
+        return obj
```

Comment:
not make a sense. Please check how this method implemented in the DRF lib

Reply:
Changed to lookup_url_kwarg